js/kucoinfutures.js

```diff
@@ -0,0 +1,1287 @@
+            // 'cancelAfter': long, // cancel after n seconds, requires timeInForce to be GTT
+            // 'postOnly': false, // Post only flag, invalid when timeInForce is IOC or FOK
+            // 'hidden': false, // Order will not be displayed in the order book
+            // 'iceberg': false, // Only a portion of the order is displayed in the order book
+            // 'visibleSize': this.amountToPrecision (symbol, visibleSize), // The maximum visible size of an iceberg order
+            // market orders --------------------------------------------------
+            // 'size': this.amountToPrecision (symbol, amount), // Amount in base currency
+            // 'funds': this.costToPrecision (symbol, cost), // Amount of quote currency to use
+            // stop orders ----------------------------------------------------
+            // 'stop': 'loss', // loss or entry, the default is loss, requires stopPrice
+            // 'stopPrice': this.priceToPrecision (symbol, amount), // need to be defined if stop is specified
+            // 'stopPriceType' // Either TP, IP or MP, Need to be defined if stop is specified.
+            // margin orders --------------------------------------------------
+            // 'marginMode': 'cross', // cross (cross mode) and isolated (isolated mode), set to cross by default, the isolated mode will be released soon, stay tuned
+            // 'autoBorrow': false, // The system will first borrow you funds at the optimal interest rate and then place an order for you
+            // futures orders -------------------------------------------------
+            // reduceOnly // (boolean) A mark to reduce the position size only. Set to false by default. Need to set the position size when reduceOnly is true.
+            // closeOrder // (boolean) A mark to close the position. Set to false by default. It will close all the positions when closeOrder is true.
+            // forceHold // (boolean) A mark to forcely hold the funds for an order, even though it's an order to reduce the position size. This helps the order stay on the order book and not get canceled when the position size changes. Set to false by default.
+        };
+        let amountString = this.amountToPrecision (symbol, amount);
+        request['size'] = this.amountToPrecision (symbol, amount);
+        if (uppercaseType === 'LIMIT') {
+            amountString = this.amountToPrecision (symbol, amount);
+            request['size'] = parseInt (amountString);
+            request['price'] = this.priceToPrecision (symbol, price);
+        }
+        const response = await this.futuresPrivatePostOrders (this.extend (request, params));
+        //
+        //    {
+        //        code: "200000",
+        //        data: {
+        //            orderId: "619717484f1d010001510cde",
+        //        },
+        //    }
+        //
+        const data = this.safeValue (response, 'data', {});
+        const timestamp = this.milliseconds ();
```

Comment:
we cannot make up data that is not in response

Reply:
We do this in a decent amount of other places, I can go fix them, but just letting you know it's already all over the place